File: example/java/spring-boot/one_consumer_two_providers/account-api/src/main/java/br/com/zup/pact/accountapi/stub/AccountStub.java
package br.com.zup.pact.accountapi.stub;

import br.com.zup.pact.accountapi.dto.AccountDetailsDTO;
import br.com.zup.pact.accountapi.entity.Account;
import br.com.zup.pact.accountapi.enums.AccountType;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;
import lombok.Getter;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Service
@Slf4j
public class AccountStub {

    private static final Integer NUMBER_OF_STUBS = 10;
    private static final BigDecimal INITIAL_BALANCE = new BigDecimal("100");
    private static final List<AccountType> ACCOUNT_TYPES = Arrays.asList(AccountType.class.getEnumConstants());
    @Getter
    private Map<Integer, Account> accounts;


    public AccountStub() {
        log.info("\n\n\n\t\t\t\t\t\t ============================ Creating Account Stubs! ============================ \n");
        accounts = createStubs(NUMBER_OF_STUBS);
    }

    private Map<Integer, Account> createStubs(int numberOfStubs) {
        final Map<Integer, Account> accounts = new HashMap<>(NUMBER_OF_STUBS);
        for (int i = 1; i <= numberOfStubs; i++) {
            Collections.shuffle(ACCOUNT_TYPES);
            final Account account = Account.builder()
                    .id(i)
                    .clientId(i)
                    .balance(INITIAL_BALANCE)
                    .accountType(ACCOUNT_TYPES.get(0))
                    .build();
            accounts.put(i, account);
        }
        return accounts;
    }

    public List<AccountDetailsDTO> getAllStubsDTOFormat() {
        final List<AccountDetailsDTO> clientDetailsDTOS = new ArrayList<>();
        final List<Account> accounts = this.accounts.values().stream()
                .collect(Collectors.toList());
        for (Account account : accounts) {
            clientDetailsDTOS.add(Account.fromEntityToDto(account));
        }
        return clientDetailsDTOS;
    }
}

File: example/java/kotlin/provider/src/main/kotlin/br/com/zup/pact/provider/stub/AccountStub.kt
package br.com.zup.pact.provider.stub

import br.com.zup.pact.provider.dto.AccountDetailsDTO
import br.com.zup.pact.provider.entity.AccountEntity
import br.com.zup.pact.provider.enums.AccountType
import org.springframework.stereotype.Component
import javax.annotation.PostConstruct

@Component
class AccountStub {

    private val numberOfStubs = 10
    private val initialBalance = 100.0
    private val initialAccountType = AccountType.COMMON

    val accounts = mutableMapOf<Int, AccountEntity>()

    @PostConstruct
    private fun createStubs() {
        accounts.putAll(
                (1..numberOfStubs)
                        .map { item -> AccountEntity(item, item, initialBalance, initialAccountType) }
                        .associateBy({ it.accountId }, { it })
        )
    }

    fun getAllStubsDTOFormat(): List<AccountDetailsDTO> {
       return accounts.values.map(AccountEntity::toDto)
    }

}